python/sparkdl/graph/input.py

```diff
@@ -0,0 +1,317 @@
+        The graph might contain variables that are maintained in the provided session.
+        Thus we need an active session in which the graph's variables are initialized or
+        restored. We do not close the session. As a result, this constructor can be used
+        inside a standard TensorFlow session context.
+
+        .. code-block:: python
+
+             with tf.Session() as sess:
+                  graph = import_my_tensorflow_graph(...)
+                  TFInputGraph.fromGraph(graph, sess, ...)
+
+        :param graph: `tf.Graph`
+        :param feed_names: list, names of the input tensors.
+        :param fetch_names: list, names of the output tensors.
+        """
+        return _build_with_feeds_fetches(sess=sess, graph=graph, feed_names=feed_names,
+                                         fetch_names=fetch_names)
+
+    @classmethod
+    def fromGraphDef(cls, graph_def, feed_names, fetch_names):
+        """
+        Construct a TFInputGraph from a tf.GraphDef object.
+
+        :param graph_def: `tf.GraphDef`, a serializable object containing the topology and
+                           computation units of the TensorFlow graph.
+        :param feed_names: list, names of the input tensors.
+        :param fetch_names: list, names of the output tensors.
+        """
+        assert isinstance(graph_def, tf.GraphDef), \
+            ('expect tf.GraphDef type but got', type(graph_def))
+
+        graph = tf.Graph()
+        with tf.Session(graph=graph) as sess:
+            tf.import_graph_def(graph_def, name='')
+            gin = _build_with_feeds_fetches(sess=sess, graph=graph, feed_names=feed_names,
+                                            fetch_names=fetch_names)
+        return gin
+
+    @classmethod
+    def fromCheckpoint(cls, checkpoint_dir, feed_names, fetch_names):
+        """
+        Construct a TFInputGraph object from a checkpoint, ignore the embedded
+        signature_def, if there is any.
+
+        :param checkpoint_dir: str, name of the directory containing the TensorFlow graph
+                               training checkpoint.
+        :param feed_names: list, names of the input tensors.
+        :param fetch_names: list, names of the output tensors.
+        """
+        return _from_checkpoint_impl(checkpoint_dir, signature_def_key=None, feed_names=feed_names,
+                                     fetch_names=fetch_names)
+
+    @classmethod
+    def fromCheckpointWithSignature(cls, checkpoint_dir, signature_def_key):
+        """
+        Construct a TFInputGraph object from a checkpoint, using the embedded
+        signature_def. Throw error if we cannot find an entry with the `signature_def_key`
+        inside the `signature_def`.
+
+        :param checkpoint_dir: str, name of the directory containing the TensorFlow graph
+                               training checkpoint.
+        :param signature_def_key: str, name of the mapping contained inside the `signature_def`
```

Comment:
we don't need to say we retrieve the mapping here since that is not always relevant to the user (and is really an implementation detail). also that is not the only thing we do - we use it to get the fetches to freeze the graph. should be sufficient to say: "key (name) of the signature_def to use. It should be in the list of signature_def's saved with the checkpoint.".